docs/research/format_to_parts.md

```diff
@@ -0,0 +1,190 @@
+Format to Parts in ICU4X
+========================
+
+ECMA-402 specifies *[formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts#Description)*, a variant of the *format* function that returns the string split into segments according to what the segment represents.  For example:
+
+```javascript
+new Intl.DateTimeFormat("fr").format(new Date())
+// "19/11/2020"
+
+new Intl.DateTimeFormat("fr").formatToParts(new Date())
+/* [
+  {type: "day", value: "19"},
+  {type: "literal", value: "/"},
+  {type: "month", value: "11"},
+  {type: "literal", value: "/"},
+  {type: "year", value: "2020"},
+] */
+
+new Intl.DateTimeFormat("fr", { dateStyle: "medium" }).formatRange(new Date(2020, 0, 1), new Date(2020, 2, 3))
+// "1 janv. – 3 mars 2020"
+
+new Intl.DateTimeFormat("fr", { dateStyle: "medium" }).formatRangeToParts(new Date(2020, 0, 1), new Date(2020, 2, 3))
+/* [
+  {type: "day", value: "1", source: "startRange"},
+  {type: "literal", value: " ", source: "startRange"},
+  {type: "month", value: "janv.", source: "startRange"},
+  {type: "literal", value: " – ", source: "shared"},
+  {type: "day", value: "3", source: "endRange"},
+  {type: "literal", value: " ", source: "endRange"},
+  {type: "month", value: "mars", source: "endRange"},
+  {type: "literal", value: " ", source: "shared"},
+  {type: "year", value: "2020", source: "shared"},
+] */
+```
+
+The goal of this document is to discuss how we will deal with formatToParts in ICU4X, both with regard to the API model and the data model (implementation).
```

Comment:
I changed "API model" to "external data model" since I'm not actually talking about an API. This is intended to talk abstractly about the *capabilities* of a formatted string.